### PR TITLE
Release 0.26.0

### PR DESCRIPTION
**Pull request by bcardiff**
Preview at https://github.com/bcardiff/crystal/blob/changelog/0.26.0/CHANGELOG.md

Note: I still need to convert PR references to links and there are a couple of PR to be merged before this.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,131 @@
...
+- Refactor codegen of binary operators. ([#6330], thanks @bcardiff)
+- Refactor use `JSON::Serializable` instead of `JSON.mapping`. ([#6308], thanks @kostya)
+- Change how metaclasses are shown. Use `Foo.class` instead of `Foo:Class`. ([#6439], thanks @RX14)
+
+## Tools
+
+### Formatter
+
+- Fixed formatting of `{ {1}.foo, ...}` like expressions. ([#6300], thanks @asterite)
+- Fixed formatting of when with numbers. Use right alignment only if all are number literals. ([#6392], thanks @MakeNowJust)
```
Comment (straight-shoota) [inline CHANGELOG.md]: `when` in backticks

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,131 @@
...
+- **(breaking-change)** Constant lookup context in macro is now lexical. ([#5354], thanks @MakeNowJust)
+- **(breaking-change)** Evaluate instance var initializers at the metaclass level (ie: disallow using `self`). ([#6414], thanks @asterite)
+- Add `&+` `&-` `&*` `&**` operators parsing. NB: No behaviour is assigned to these operators yet. ([#6329], thanks @bcardiff)
+- Add support for empty `case` without `when`. ([#6367], thanks @straight-shoota)
+
+## Standard library
+
+### Macros
+
+- Add `pp!` and `p!` macro methods. ([#6374], thanks @straight-shoota)
```
Comment (bew) [inline CHANGELOG.md]: I don't think this change lies in the standard library section, as you can't add or change the available macro methods

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,131 @@
...
+
+### Playground
+
+## Others
+
+- Fixed `system_spec` does no longer emit errors messages on BSD platforms. ([#6289], thanks @jcs)
+- Fixed compilation issue when running spec against compiler and std together. ([#6312], thanks @straight-shoota)
+- Add support for llvm 6.0. ([#6381], [#6380], [#6383], thanks @felixbuenemann)
+- CI improvements and housekeeping. ([#6313], [#6337], [#6407], [#6408] thanks @bcardiff, @MakeNowJust, @r00ster91)
+- Flatten project structure created by crystal init. ([#6317], thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md]: Shouldn't this belong to *Tools*?

**Comment by straight-shoota** [comment]
I think the last two changes belong to *Standard Library*. And maybe mention that #6333 also added an optional argument to `String#check_no_null_byte`.

File: CHANGELOG.md (hunk 2 above)
Comment (straight-shoota) [reply CHANGELOG.md]: True. But macros are also documented in stdlib API. I'm not sure about this.

But it reminds me that I actually forgot adding docs for `p! and `pp!` in that PR :o

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,131 @@
...
+
+- Add line numbers at link when there are duplicated filenames in "Defined in:" section. ([#6280], thanks @r00ster91)
+
+### Playground
+
+## Others
+
+- Fixed `system_spec` does no longer emit errors messages on BSD platforms. ([#6289], thanks @jcs)
+- Fixed compilation issue when running spec against compiler and std together. ([#6312], thanks @straight-shoota)
+- Add support for llvm 6.0. ([#6381], [#6380], [#6383], thanks @felixbuenemann)
```
Comment (j8r) [inline CHANGELOG.md]: LLVM is normally in uppercase

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,131 @@
...
+- CI improvements and housekeeping. ([#6313], [#6337], [#6407], [#6408] thanks @bcardiff, @MakeNowJust, @r00ster91)
+- Flatten project structure created by crystal init. ([#6317], thanks @straight-shoota)
+- Refactor platform specifics from `ENV` to `Crystal::System::Env` and implement for windows. ([#6333], thanks @straight-shoota)
+- Add support for `WinError` UTF16 string messages. ([#6442], thanks @straight-shoota)
```
Comment (j8r) [inline CHANGELOG.md]: UTF-16 with a dash is the standard naming, to be consistent with the UTF-8 written above

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,131 @@
+# 0.26.0 (2018-08-??)
+
+## Language changes
+
+- **(breaking-change)** Revert do not collapse unions for sibling types. ([#6351], thanks @asterite)
+- **(breaking-change)** Restrict `exp.@var` to types in the same hierarchy. ([#6358], thanks @asterite)
+- **(breaking-change)** Constant lookup context in macro is now lexical. ([#5354], thanks @MakeNowJust)
+- **(breaking-change)** Evaluate instance var initializers at the metaclass level (ie: disallow using `self`). ([#6414], thanks @asterite)
+- Add `&+` `&-` `&*` `&**` operators parsing. NB: No behaviour is assigned to these operators yet. ([#6329], thanks @bcardiff)
+- Add support for empty `case` without `when`. ([#6367], thanks @straight-shoota)
```
Comment (j8r) [inline CHANGELOG.md]: We could precise the main purpose of this: ... "to generate them with macros"

Comment (straight-shoota) [reply CHANGELOG.md]: An explanation for any change can be found in the respective PRs. It's unnecessary to mention here.

**Comment by wontruefree** [comment]
Are there still concerns about this PR?
https://github.com/crystal-lang/crystal/pull/6358

**Comment by Qwerp-Derp** [comment]
Is #6419 going to be included in 0.26.0 or is that going to have to wait until the next release?

File: CHANGELOG.md (hunk 2 above)
Comment (bcardiff) [reply CHANGELOG.md]: Yeah, I'll move it to language changes. Macro methods are defined in the language itself at the end of the day. But I will keep a Macros subsection for that.

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,137 @@
...
+## Standard library
+
+### Numeric
+
+- Fixed `Random#rand(Range(Float, Float))` to return `Float`. ([#6445], thanks @straight-shoota)
+- Add docs of big module overloads. ([#6336], thanks @laginha87)
+
+### Text
+
+- **(breaking-change)** `String#from_utf-16(pointer : Pointer(UInt16))` returns now `{String, Pointer(UInt16)}`. ([#6333], thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md]: One grep to much ;)

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,137 @@
...
+### Playground
+
+## Others
+
+- Fixed `system_spec` does no longer emit errors messages on BSD platforms. ([#6289], thanks @jcs)
+- Fixed compilation issue when running spec against compiler and std together. ([#6312], thanks @straight-shoota)
+- Add support for LLVM 6.0. ([#6381], [#6380], [#6383], thanks @felixbuenemann)
+- CI improvements and housekeeping. ([#6313], [#6337], [#6407], [#6408], [#6315], thanks @bcardiff, @MakeNowJust, @r00ster91, @maiha)
+- Refactor platform specifics from `ENV` to `Crystal::System::Env` and implement for windows. ([#6333],
+- Add an optional argument to `String#check_no_null_byte.thanks @straight-shoota`)
```
Comment (straight-shoota) [inline CHANGELOG.md]: This change is already listed in *Text*, but the one above misses the attribution.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,137 @@
...
+- **(breaking-change)** Remove `File.each_line` method that returns an iterator. Use `IO#each_line`. ([#6301], thanks @asterite)
+- Fixed `File.join` when path separator is a component argument. ([#6328], thanks @icyleaf)
+- Fixed `Dir.glob` can now list broken symlinks. ([#6466], thanks @straight-shoota)
+- Add `File` and `Dir` support for windows. ([#5623], thanks @RX14)
+
+### Networking
+
+- Fixed `Socket#accept?` base implementation. ([#6277], thanks @ysbaddaden)
+- Fixed performance issue due to unbuffered `IO` read. `IO#sync` only affect writes, introduce `IO#read_buffering?`. ([#6304], [#6474], thanks @asterite, @bcardiff)
+- Fixed hanlding of closed state in `HTTP::Server::Response`. ([#6477], thanks @straight-shoota)
```
Comment (straight-shoota) [inline CHANGELOG.md]: "handling"

Comment (bcardiff) [reply CHANGELOG.md]: How did the spell checker turned off? 😅

File: CHANGELOG.md (hunk 7 above)
Comment (bcardiff) [reply CHANGELOG.md]: Am I missing something or you are pointing the fact that is was duplicated?

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,136 @@
+# 0.26.0 (2018-08-??)
+
+## Language changes
+
+- **(breaking-change)** Revert do not collapse unions for sibling types. ([#6351], thanks @asterite)
+- **(breaking-change)** Restrict `exp.@var` to types in the same hierarchy. ([#6358], thanks @asterite)
```
Comment (luislavena) [inline CHANGELOG.md]: Due recent revert #6484, this is no longer true 😢

Comment (bcardiff) [reply CHANGELOG.md]: That's right, but don't be sad :-).

FYI, We are waiting a couple of more PR and small fixes before a final round on the changelog is done and the release process begin.

**Comment by Qwerp-Derp** [comment]
Windows should be uppercase, not lowercase.

**Comment by bcardiff** [comment]
@wdhwg001 it was, but I double check with in other places in docs and lower case was used.

**Pull request merged**